mock_ticket_api/utils.py

```diff
@@ -58,3 +79,114 @@ def get_confirmations_data(pk, include_qr=True) -> dict:
             }
         ],
     }
+
+
+def get_actual_confirmations_data(pk, include_qr=True) -> dict:
+    if include_qr:
+        path = PurePath(__file__).parent.joinpath("data", "ticket_qr.png")
+        with open(path.as_posix(), "rb") as f:
+            ticket = f"data:image/png;base64,{base64.b64encode(f.read()).decode()}"
+    else:
+        ticket = "QR_CODE"
+
+    valid_from = timezone.now()
+    vat = Decimal("0.24")
+
+    booking = Booking.objects.get(source_id=pk)
+    feed = booking.ticketing_system.feeds.first()
+    route = Route.objects.get(translations__long_name=booking.route_name, feed=feed)
+    agency = route.agency
+    departure = Departure.objects.filter(trip__route=route).first()
+    from_stop_time = departure.trip.stop_times.order_by("stop_sequence").first()
+    to_stop_time = departure.trip.stop_times.order_by("stop_sequence").last()
+    fare = route.fares.first()
+    rider_category = fare.rider_categories.first()
+    block_ids = route.trips.values_list("block_id", flat=True).distinct()
+
+    return {
+        "id": pk,
+        "status": "CONFIRMED",
+        "tickets": [
+            {
+                "id": uuid.uuid4(),
+                "agency": {
+                    "name": get_translated_field(agency, "name"),
+                    "logo_url": agency.logo_url,
+                },
+                "customer_type": {
+                    "description": get_translated_field(rider_category, "description"),
+                    "name": get_translated_field(rider_category, "name"),
+                },
+                "locale": get_language(),
+                "maas_operator_id": booking.maas_operator.identifier,
+                "price": {
+                    "amount_excluding_vat": format_price(
+                        FareRiderCategory.objects.get(
+                            fare=fare, rider_category=rider_category
+                        ).price
+                        / (1 + vat)
+                    ),
+                    "amount_total": format_price(fare.price),
```

Comment:
Prices seem to be defined as numbers in the specification. Here they are strings, but maybe that is fine.

Reply:
Yeah, good point. IMHO the prices should be strings, so it is the specification that needs changes 🙂